airflow/www/security.py

```diff
@@ -453,12 +453,19 @@ def merge_pv(perm, view_menu):
 
     def update_admin_perm_view(self):
         """
-        Admin should have all the permission-views.
+        Admin should have all the permission-views, except the dag views.
+        because Admin have already have all_dags permission.
         Add the missing ones to the table for admin.
 
         :return: None.
         """
-        pvms = self.get_session.query(sqla_models.PermissionView).all()
+        all_dag_view = self.find_view_menu('all_dags')
+        dag_perm_ids = [self.find_permission('can_dag_edit').id, self.find_permission('can_dag_read').id]
+        pvms = self.get_session.query(sqla_models.PermissionView).filter(~and_(
+            sqla_models.PermissionView.permission_id.in_(dag_perm_ids),
+            sqla_models.PermissionView.view_menu_id != all_dag_view.id)
+        ).all()
```

Comment:
Why not directly use `self.find_permission_view_menu` if you are simply looking for the `pvm` for `all_dags`?

Reply:
@XD-DENG I need to take a deeper look, but the initial idea is to add all the permissions with views to Admin(not just those dag related). If we are going to add single view-menu(`all_dags`) to Admin, we need to add the rest to Admin(or it has been handled inside FAB which I am not sure) as well otherwise  Admin will miss bunches of permissions staffs.

Reply:
Thanks @feng-tao for the clarification :)

Reply:
Are you comment on line 464-467? These code is not looking for `pvm` for `all_dags`
These SQL code intent to filter out `can_dag_read/edit` on all dag views, and keep `can_dag_read/edit` on `all_dags`

Please notice the `~and_` in line 464 and the `!=` in line 466